### PR TITLE
(Astro rework) Redesign of /community page and its subpages

### PR DESCRIPTION
**Pull request by kacperwyczawski**
Changes:
- migrate `/community`, `/community/[team]`, `/community/event-funding` and `/community/commercial-support` to Astro
- update DaisyUI to `4.0.4` for timeline support
- animate RFC diagram
- load content from content collections (22 new `.yaml` or `.mdx` files)
- introduce `MDXLayout` for creating pages in mdx
- translate content from `html` to `mdx`/`yaml`
- use Fisher-Yates algorithm for unbiased shuffle in commercial support page
- add members of nixcon team, based on [this post](https://discourse.nixos.org/t/governance-through-project-leads-in-nixcon-2024/33981#how-can-you-get-involved-3)
- add contact section to team pages
- update link for adding company to `/commercial-support`, will be active after merge

What can be improved in another PR:
- #1162 
- #1161

### CODE REVIEW

**Review by garbas — APPROVED** [review]
Looks good, but I'd like @thilobillerbeck to also look at it.

Would you also create issues for things you've listed under "What can be improved in another PR"?

**Comment by thilobillerbeck** [comment]
I'll have a look at it later on. Thx for working on that. Regarding your improvement ideas:

> * theming and colors using [DaisyUI themes](https://daisyui.com/docs/themes/#-4); [benefits](https://daisyui.com/docs/colors/#-1) (out of scope for this PR)

At the current state I'm unsure how much DaisyUI will be even required in the final result and I'm unsure if it's better to just omit it if it won't be needed that much or at all.

> * better support for dark theme (depends on above)

Definitely a good point, but currently I think we should work on migrating everything as is. But this is one of the primary things that goes through my head all the time.

> * use `<Picture>` component, requires updating astro to `3.3`. See [those release notes](https://astro.build/blog/astro-330/). This applies to other pages as well

I already have a branch to improve things on tailwind defaults to comply more with the currently live version, I guess I can include an update to astro as well.

I'll add more when I do the review. :)

File: src/layouts/MDXLayout.astro
Comment (thilobillerbeck) [inline src/layouts/MDXLayout.astro:1]: This has quite a few differences to the current design.
image
image

- Members should be in a box on the right side and not inline with the infotext (you can use the metadata section in the markdown file for that)
- The further links were originally rendered as buttons which would be great to see here as well

File: src/pages/community/teams/[...slug].astro
Comment (thilobillerbeck) [inline src/pages/community/teams/[...slug].astro:1]: The comment from MDXLayout applies here as well

File: src/pages/community/commercial-support.astro
Comment (thilobillerbeck) [inline src/pages/community/commercial-support.astro:1]: Please bring this in line with the current design
image

This means:
- Putting every supporter in its distinct box
- making the image sizes equal
- make fonts and colors comply with the rest of the site

**Review by thilobillerbeck — CHANGES_REQUESTED** [review]
First of all thanks for handing in your changes. <3

As a heads up for this review I want to refer to #1125 which describes our current goal for the redesign. I know you have many ideas but please keep in mind that moving the tech stack over to Astro is currently the number one priority.

If you have any questions just ask. :)

**Comment by kacperwyczawski** [comment]
Oops, I didn't know about #1125. Alright then, lets start with the changes.

File: src/styles/base.css [hunk 1]
```diff
@@ -31,3 +31,106 @@
...
+    .rfc-box-3 {
+        animation: 10s linear infinite rfc-box-3
+    }
+    .rfc-box-4 {
+        animation: 10s linear infinite rfc-box-4
+    }
+    .rfc-box-x {
+        animation: 10s linear infinite rfc-box-x
+    }
+}
```
[the remark this answers is not in the document]
Comment (kacperwyczawski) [reply src/styles/base.css:136]: Indeed, scroll trigger would be nice, I'll implement it

File: src/pages/community.astro [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        <div class="border-2 border-gray-300 bg-gray-100 rounded-xl px-4 py-4 text-lg w-64 rfc-box-4 relative ml-12">
+          <svg class="inline relative stroke-base-content -top-0.5" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-file-check-2"><path d="M4 22h14a2 2 0 0 0 2-2V7.5L14.5 2H6a2 2 0 0 0-2 2v4"/><polyline points="14 2 14 8 20 8"/><path d="m3 15 2 2 4-4"/></svg>
+          Submission accepted
+          <svg class="absolute w-8 h-8 -left-[calc(2rem_+_2px)] top-0" width="30" height="30" viewBox="0 0 30 30" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path class="stroke-2 stroke-gray-300" d="M30 29.5C26.126 29.5 22.2899 28.737 18.7108 27.2544C15.1317 25.7719 11.8797 23.599 9.14035 20.8596C6.40102 18.1203 4.22807 14.8683 2.74555 11.2892C1.26304 7.71005 0.5 3.87399 0.5 -2.57897e-06"/>
+          </svg>
+          <hr class="absolute border border-gray-300 h-[calc(2rem_+_2px)] w-0 -left-[calc(2rem_+_2px)] -top-[calc(2rem_+_2px)]"/>
+        </div>
+      </div>
+    </div>
```
[the remark this answers is not in the document]
Comment (kacperwyczawski) [reply src/pages/community.astro]: > the animation lacks some steps from the original graph which strips some context away which should be there to completely understand the review process.

Is this okay?

image

Or should I add more missing nodes (`Refine submission`, `New submission` and `Substantial new arguments`)? I'm asking, because it would require changes to graph introduced in this PR. Old graph for reference:

image

File: src/pages/community.astro [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+                <span class="block font-bold group-hover:underline">
+                  Read more
+                </span>
+              </p>
+            </a>
+          </li>
+        ))}
+      </ul>
+    </Container>
+  </div>
```
[the remark this answers is not in the document]
Comment (kacperwyczawski) [reply src/pages/community.astro:248]: > Please use the icons from the current page for the teams.

What about *documentation* and *nixpkgs architecture* teams? They don't have icons on current website. Should I leave an empty space/placeholder?

File: src/layouts/MDXLayout.astro
Comment (kacperwyczawski) [reply src/layouts/MDXLayout.astro:1]: > Members should be in a box on the right side and not inline with the infotext (you can use the metadata section in the markdown file for that)

Fixed in [`1cad122`](https://github.com/NixOS/nixos-homepage/pull/1160/commits/1cad122dcb5cbad1cfc3462af6e702a4f7c35404)

> The further links were originally rendered as buttons which would be great to see here as well

What do you think about putting those links under *Members*? Like in current [marketing team's page](https://nixos.org/community/teams/marketing).

**Comment by thilobillerbeck** [comment]
@kacperwyczawski how's progress going? Do you have a rough estimation of when you'll be finished?

**Comment by kacperwyczawski** [comment]
@thilobillerbeck can you please address [this](https://github.com/NixOS/nixos-homepage/pull/1160#discussion_r1405456252), [this](https://github.com/NixOS/nixos-homepage/pull/1160#discussion_r1412792196) and [this](https://github.com/NixOS/nixos-homepage/pull/1160#discussion_r1405455377). Thanks in advance. Besides these things, everything is finished.

File: src/pages/community.astro (hunk 3 above)
[the remark this answers is not in the document]
Comment (thilobillerbeck) [reply src/pages/community.astro]: I guess some kind of placholder would be good, maybe the white Nix Flake for example

File: src/layouts/MDXLayout.astro
Comment (thilobillerbeck) [reply src/layouts/MDXLayout.astro:1]: yeah, I think bringing that in line with the current marketing page is the way to go, since the buttons look a bit out of place anyway imo.

File: src/pages/community.astro (hunk 2 above)
[the remark this answers is not in the document]
Comment (thilobillerbeck) [reply src/pages/community.astro]: The difference between your graph and the one currently used is, that the currently used one at a glance and without an animation shows which paths the RFC process can take and can also be percieved by people who, for example, have a reduced motion preference set in their browser, since they likely wont get to see the animation and cannot percieve the whole context this image should offer. Apart from that, the animation and the graph you submitted lack the steps of "Refine..." and "New...", which make a big difference in how submissiosn are handeled between processes. 

So tl:dr; yes, please include all branches into your graph since they provide a version of the graph that is percieveable by every visitor. Animations are fine, but they should just enhance the context and not communicate crucial context of a graph.

**Comment by kacperwyczawski** [comment]
I had to make more changes to the diagram animation and the diagram itself, than I thought. I'll finish it tomorrow, sorry for delay.

**Comment by kacperwyczawski** [comment]
@thilobillerbeck I'm ready for another review :rocket:

**Review by thilobillerbeck — APPROVED** [review]
Looks good overall, thx for the contributions. There are some minor nitpicks I have, but to save time on communication I will fix them myself. 

image
Unfortunately I'm not really happy with the NixCon section, it looks crammed together now, has thick surrounding scrollbars and does not feature the old descriptions. I will roll this back to the old design but encourage you to iterate on the idea since it is interesting, it's (apart being not in the scope for this stage of the project) not really ready imo. Please don't take this personal, we just want to get this done til the end of the year and can still incorporate this in the next stage.

Anyway, thanks for the contributions and the ideas. <3

**Comment by kacperwyczawski** [comment]
No problem, I understand your decision with nixcons section. I'm glad I could help.

**Comment by Aleksanaa** [comment]
@kacperwyczawski Sorry to bother,

**Comment by Aleksanaa** [comment]
@kacperwyczawski Sorry to bother, I think I missed some context in this discussion. Where did the meetup list come from? Some of them seem to be invalid.

**Comment by kacperwyczawski** [comment]
> @kacperwyczawski Sorry to bother, I think I missed some context in this discussion. Where did the meetup list come from? Some of them seem to be invalid.

When I migrated this page to Astro, I left the content of the meetups section untouched. Here is the relevant code from the previous commit:

https://github.com/NixOS/nixos-homepage/blob/d36a631930ab7dcb831c6ed6848d8a060f70774d/community/index.tt#L196-L232

If you want to fix/remove the invalid ones, just edit this file: https://github.com/NixOS/nixos-homepage/blob/main/core/src/content/community/meetups.yaml. I hope this helps!
